kasa/iot/iottimezone.py

```diff
@@ -0,0 +1,156 @@
+"""Module for io device timezone lookups."""
+
+from __future__ import annotations
+
+import logging
+from datetime import datetime, tzinfo
+
+from zoneinfo import ZoneInfo
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_timezone(index: int) -> tzinfo:
```

Comment:
I'm wondering if/where should we cache this? Maybe lru_cache here?

Reply:
Seems unnecessary for a dict lookup no?

Reply:
It'd rather be to avoid constructing new `ZoneInfo` instances.

Reply:
So I've implemented a non-expiring cache in [this commit](https://github.com/python-kasa/python-kasa/pull/1147/commits/8a2398dbd8469a75f4a1b7b40ba5dd69cfb322ed). I think non-expiring should be fine as there's only a fixed number of firmware indexes and it's unlikely more than a few would actually be called. It also creates the `ZoneInfo` in an executor thread due to [this post](https://developers.home-assistant.io/blog/2024/05/19/fix_zoneinfo_blocking_io)